### ISSUE TITLE
关于_ds_init_train_model的疑问

### ISSUE THREAD

**Issue by BeerTai** [OpenRLHF/OpenRLHF#495]
在_ds_init_train_model.py中，如果 is_actor是true，则model.model = engine，否则model = engine
然后返回的是model，是不是意味着，如果是actor类，经过deepspeed初始化后，返回的model.model才是真正的模型？如果是这样的化，为什么在dpo代码里在设置train()和eval()时都是针对model.train()以及model.eval()? 而不是model.model.trian()以及model.model.eval()

**Comment by hijkzzz** [comment]
对的 Actor 内部包装一个 Engine
而 Reward Model 和 Critic 本身就是一个 Engine
.train() 函数会被递归调用的
这其实是因为对于 Actor 我们想保存 model.model 到 checkpoint 以及最后的 hf model （保持原来的模型结构）
对于 Critic 直接保存其本身就好了

**Comment by BeerTai** [comment]
明白了，感谢！
